Source/Core/Core/InputSuppressor.h

```diff
@@ -0,0 +1,15 @@
+// Copyright 2025 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+class InputSuppressor final
+{
+public:
+  InputSuppressor();
+  ~InputSuppressor();
```

Comment:
The "[rule of three](https://en.cppreference.com/w/cpp/language/rule_of_three.html)" applies here.

It looks like an inadvertent copy would currently not not increase `AddSuppression`, but that object would still `RemoveSuppression` on destruction, messing up `s_suppression_count`.

Copies (and optionally moves) should be either implemented or deleted.